pwndbg/gdblib/disasm/aarch64.py

```diff
@@ -12,7 +12,45 @@
 import pwndbg.gdblib.memory
 import pwndbg.gdblib.regs
 from pwndbg.emu.emulator import Emulator
+from pwndbg.gdblib.disasm.instruction import ALL_JUMP_GROUPS
+from pwndbg.gdblib.disasm.instruction import InstructionCondition
 from pwndbg.gdblib.disasm.instruction import PwndbgInstruction
+from pwndbg.gdblib.disasm.instruction import boolean_to_instruction_condition
+
+
+def resolve_condition(condition: int, cpsr: int) -> InstructionCondition:
+    """
+    Given a condition and the NZCV flag bits, determine when the condition is satisfied
+
+    The condition is a Capstone constant
+    """
+
+    n = (cpsr >> 31) & 1
+    z = (cpsr >> 30) & 1
+    c = (cpsr >> 29) & 1
+    v = (cpsr >> 28) & 1
+
+    condition = {
+        ARM64_CC_INVALID: True,  # Capstone uses this code for the 'B' instruction, the unconditional branch
+        ARM64_CC_EQ: z == 1,
+        ARM64_CC_NE: z == 0,
+        ARM64_CC_HS: c == 1,
+        ARM64_CC_LO: c == 0,
+        ARM64_CC_MI: n == 1,
+        ARM64_CC_PL: n == 0,
+        ARM64_CC_VS: v == 1,
+        ARM64_CC_VC: v == 0,
+        ARM64_CC_HI: c == 1 and z == 0,
+        ARM64_CC_LS: not (c == 1 and z == 0),
+        ARM64_CC_GE: n == v,
+        ARM64_CC_LT: n != v,
+        ARM64_CC_GT: z == 0 and n == v,
+        ARM64_CC_LE: not (z == 0 and n == v),
+        ARM64_CC_AL: True,
+        ARM64_CC_NV: True,
+    }.get(condition, False)
```

Comment:
Fwiw Python will compute all the conditions here and construct a dict each time this is called :(

Reply:
Could be nice to refactor this to `if else ..` stuff.

Reply:
From now on I can implement this type of pattern as a `IF/ELIF` chain - eventually when we move to Python 3.10 we can move to the nicer `match/case` syntax.

Alternatively, this could be defined as a global dictionary of lambdas with each function taking the same parameters, then we lookup the function and call it so we only resolve the one condition we really care about.

Do you think this second option might be cleaner?